neo4j/templates/_imagePullSecret.tpl

```diff
@@ -17,61 +17,61 @@ imagePullSecrets:
 {{- end }}
 
-{{/* checks and throws error if registry ,username , email and name imageCredentials fields are empty */}}
-{{- define "neo4j.imageCredentials.checkForEmptyFields" -}}
+{{/*checkForEmptyFieldsAndDuplicates checks whether the imageCredential fields are empty or have any duplicates or not */}}
+{{- define "neo4j.imageCredentials.checkForEmptyFieldsAndDuplicates" -}}
 
     {{- if .Values.image.imageCredentials -}}
         {{- $errorList := list -}}
+        {{- $nameList := list -}}
         {{- range $index, $element := .Values.image.imageCredentials }}
 
-            {{- if  empty ($element.registry | trim) -}}
-                {{- $errorList = append $errorList (printf "Registry field cannot be empty for imageCredential \"%s\"" $element.name) -}}
-            {{- end -}}
-
-            {{- if  empty ($element.username | trim) -}}
-                {{- $errorList = append $errorList (printf "Username field cannot be empty for imageCredential \"%s\"" $element.name) -}}
-            {{- end -}}
+            {{- if kindIs "invalid" $element -}}
+                {{- $errorList = append $errorList (printf "Empty ImageCredential. Please provide either secretname or (registry|username|email|password|name) field(s)") -}}
+
+            {{/* check for empty registry,username,email,password,name fields only when secret name is NOT PROVIDED */}}
+            {{- else -}}
+                {{- $message := dict "fieldName" "registry" "value" $element.registry | include "neo4j.imageCredentials.checkMissingorEmptyField" -}}
+                {{- if not (empty $message) -}}
+                    {{- $errorList = append $errorList $message -}}
+                {{- end -}}
+
+                {{- $message = dict "fieldName" "username" "value" $element.username | include "neo4j.imageCredentials.checkMissingorEmptyField" -}}
+                {{- if not (empty $message) -}}
+                    {{- $errorList = append $errorList $message -}}
+                {{- end -}}
+
+                {{- $message = dict "fieldName" "password" "value" $element.password | include "neo4j.imageCredentials.checkMissingorEmptyField" -}}
+                {{- if not (empty $message) -}}
+                    {{- $errorList = append $errorList $message -}}
+                {{- end -}}
+
+                {{- $message = dict "fieldName" "name" "value" $element.name | include "neo4j.imageCredentials.checkMissingorEmptyField" -}}
+                {{- if not (empty $message) -}}
+                    {{- $errorList = append $errorList $message -}}
+                {{- else -}}
+                    {{- $nameList = append $nameList ($element.name | trim) -}}
+                {{- end -}}
+
+                {{- $message = dict "fieldName" "email" "value" $element.email | include "neo4j.imageCredentials.checkMissingorEmptyField" -}}
+                {{- if not (empty $message) -}}
+                    {{- $errorList = append $errorList $message -}}
+                {{- end -}}
 
```

Comment:
the fields are added based on the official documentation here
https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/#create-a-secret-by-providing-credentials-on-the-command-line
since the secret as mentioned requires username , password , email and registry i dont think we should make it optional...i can try if it works with just username or email but registry should not be optional as it can differ from docker hub to any other cloud provider registry

Reply:
tried and it works with registry and email..hence made both of them optional